src/dvc_objects/db.py

```diff
@@ -142,6 +142,13 @@ def add(
                 callback=Callback.as_callback(cb),
             )
 
+    def delete(self, oid: str):
+        self.fs.remove(self.oid_to_path(oid))
+
+    def clear(self):
+        for oid in self.all():
```

Comment:
`self.all()` is terribly slow. I know this is just a convenience method, but it adds up (even for mnist).

Reply:
Also would be nice to have `dvc-data clear` CLI command and reuse `odb.delete` in `dvc-data rm`.

Reply:
Yeah, performance doesn't matter here for now.

Regarding the cli, I'm not using it and there are no tests, so it didn't even cross my mind to touch it this time. Could be added any time later.

Reply:
I was using `self.all()` in `exists_prefix` and had to optimize. https://github.com/iterative/dvc-objects/commit/ebee28733103ac2b13c5207f640394ff03603699

Reply:
> Could be added any time later.

It's not here, it's in dvc-data.

Reply:
https://github.com/iterative/dvc-data/pull/175 @skshetry